pkg/vault/vault.go

```diff
@@ -75,26 +89,36 @@ func NewVault(cfg Config) (*Vault, error) {
 		return nil, err
 	}
 
-	authMethod, err := cfg.Auth.authMethod()
-	if err != nil {
-		return nil, err
-	}
-
-	authFac := authFactoryReal{}
-	_, err = authMethod.authenticate(context.Background(), &authFac, client)
+	authToken, err := getAuthToken(context.Background(), &cfg.Auth, client)
 	if err != nil {
-		return nil, fmt.Errorf("error authenticating to vault: %w", err)
+		return nil, fmt.Errorf("failed to get auth token from vault: %v", err)
 	}
 
 	vault := &Vault{
-		KVStore: client.KVv2(cfg.MountPath),
+		kvStore: client.KVv2(cfg.MountPath),
+		auth:    cfg.Auth,
+		token:   authToken,
+		client:  client,
+		logger:  l,
+		authTotal: promauto.With(registerer).NewCounter(prometheus.CounterOpts{
+			Name: "cortex_vault_auth_total",
```

Comment:
> Total number of times authentication to Vault happened during token lifecycle management

I'm not sure this description is correct. It's only how many times we *started* renewal process, but we don't know how many authentication attempts there were.

From user's perspective, how is this metric different from `authLeaseRenewalSuccessTotal`?

I wonder if we should have a single "gauge" metric that's set to 1 as long as renewal is running, and we drop `cortex_vault_auth_total` and `cortex_vault_token_lease_renewal_failure_total`.
 metrics. (However I'd keep `cortex_vault_token_lease_renewal_success_total`)

WDYT?